julee_example/domain/document/document.py

```diff
@@ -89,7 +89,15 @@ class Document(BaseModel):
 
     # Additional data and content stream
     additional_metadata: Dict[str, Any] = Field(default_factory=dict)
-    content: ContentStream = Field(exclude=True)
+    content: Optional[ContentStream] = Field(default=None, exclude=True)
+    content_string: Optional[str] = Field(
+        default=None,
+        description="Small content as string (few KB max). Use for "
+        "workflow-generated content to avoid ContentStream serialization "
+        "issues. For larger content, ensure calling from concrete "
+        "implementations (ie. outside workflows and use-cases) and use "
+        "content field instead.",
+    )
```

Comment:
should we have some "either/or" validation logic? what does it mean if there is both a string and a stream, and they are not serialisations of each other?

Reply:
Yes, absolutely - that's there already though. I'll point it out.

Reply:
Oh - I see. Yes, I've got validation for that but it's on the memory/minio implementations, rather than being here in the domain where it belongs. Yes, let me change that.